modules/actions/workflows.go

```diff
@@ -69,6 +71,42 @@ func ListWorkflows(commit *git.Commit) (git.Entries, error) {
 	return ret, nil
 }
 
+// GetRunsWorkflowFileLink return url for the source of the workflow file for an ActionRun
+func GetRunsWorkflowFileLink(run *actions_model.ActionRun, gitRepo *git.Repository) string {
+	if run.Repo == nil || run.CommitSHA == "" {
+		return ""
+	}
+
+	commit, err := gitRepo.GetCommit(run.CommitSHA)
+	if err != nil {
+		return ""
+	}
+
+	entries, err := ListWorkflows(commit)
+	if err != nil {
+		return ""
+	}
+
+	var workflowEntry *git.TreeEntry
+	for _, entry := range entries {
+		if entry.Name() == run.WorkflowID {
+			workflowEntry = entry
+			break
+		}
+	}
+
+	if workflowEntry == nil {
+		return ""
+	}
+
+	workflowFilePath := workflowEntry.GetPathInRepo()
+	if workflowFilePath == "" {
+		return ""
+	}
+
+	return fmt.Sprintf("%s/src/commit/%s/%s", run.Repo.Link(), run.CommitSHA, workflowFilePath)
```

Comment:
It needs proper escaping, otherwise the link might be invalid

Reply:
Agreed, we can't influence the  workflow file